infrastructure/derive/src/lib.rs

```diff
@@ -33,3 +35,108 @@ pub fn derive_hashable_ordering(tokens: TokenStream) -> TokenStream {
     };
     gen.into()
 }
+
+/// This macro will provide a hasable implementation to the a given struct
+#[proc_macro_derive(Hashable, attributes(Digest, hash))]
```

Comment:
Some documentation and a quick code example of how to use the derive macro would help the reader here - over and above the unit tests